packages/api/cms-api/src/common/decorators/subject-entity.decorator.ts

```diff
@@ -1,20 +1,22 @@
-import { EntityName } from "@mikro-orm/core";
+/* eslint-disable @typescript-eslint/ban-types */
+import { EntityName, Loaded } from "@mikro-orm/core";
+import { SqlEntityRepository } from "@mikro-orm/postgresql";
 import { CustomDecorator, SetMetadata } from "@nestjs/common";
 
-export interface SubjectEntityOptions {
+export interface SubjectEntityOptions<Entity extends {}> {
     idArg?: string;
     pageTreeNodeIdArg?: string;
-}
-export interface SubjectEntityMeta {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    entity: EntityName<any>; //TODO
-    options: SubjectEntityOptions;
+    getEntity?: (repo: SqlEntityRepository<Entity>, args: any) => Promise<Loaded<Entity, never>>;
```

Comment:
:+1: generally, especially the typed repo is good!

I'd rething the method name, what do you think about `entityByArgs`?